web_validation_dialog/static/src/js/web_validation_dialog.js

```diff
@@ -0,0 +1,132 @@
+    var FormWidget = require('web.form_widgets');
+    var Dialog = require('web.Dialog');
+    var Model = require('web.DataModel');
+    var framework = require('web.framework');
+    var pyeval = require('web.pyeval');
+
+    var QWeb = core.qweb;
+    var _t = core._t;
+
+    FormWidget.WidgetButton.include({
+
+        /*
+         * @param field_manager: get field Value
+         * @param node: get node value
+         */
+
+        init: function (field_manager, node) {
+            this._super(field_manager, node);
+            this.options = pyeval.py_eval(this.node.attrs.options || '{}');
+            this.is_dialog_security =
+                this.options.security ? this.options.security : false;
+        },
+
+        /*
+         * @return this function return on_confirmed method
+         */
+
+        exec_action: function () {
+            var self = this;
+            if (self.node.attrs.confirm && self.is_dialog_security) {
+                Dialog.confirm(self, self.node.attrs.confirm, {
+                    buttons: [
+                        {
+                            text: _t("Ok"),
+                            classes: 'btn-primary',
+                            click: function () {
+                                this.close();
+                                self.open_pincode_dialog();
+                            },
+                        },
+                        {
+                            text: _t('Cancel'),
+                            close: true,
+                        },
+                    ],
+                });
+            } else if (self.node.attrs.confirm) {
+                var def = $.Deferred();
+                Dialog.confirm(self,
+                    self.node.attrs.confirm,
+                    {
+                        confirm_callback: self.on_confirmed,
+                    }
+                ).on("closed", null, function () {
+                    def.resolve();
+                });
+                return def.promise();
+            } else {
+                return self.on_confirmed();
+            }
+        },
+        execute_action: function () {
+            var self = this;
+            if (!this.node.attrs.special) {
+                return this.view.recursive_save().then(self.exec_action());
+            }
+            return self.exec_action();
+        },
+        open_pincode_dialog : function () {
+            var self = this;
+            new Dialog(self, {
+                title: _t('Validation'),
+                size : "small",
+                $content: QWeb.render('DialogValidation'),
+                buttons: [
+                    {
+                        text: _t("Ok"),
+                        classes: 'btn-primary',
+                        click: function () {
+                            var curr_obj = this;
+                            var password = this.$el.find("#pincode").val();
+                            if (password) {
```

Comment:
You can mark a field as `required="required"` on the html instead of writing code to handle it.